src/main/java/ch/unisg/ics/interactions/wot/td/clients/UriTemplate.java

```diff
@@ -0,0 +1,145 @@
+package ch.unisg.ics.interactions.wot.td.clients;
+
+import ch.unisg.ics.interactions.wot.td.schemas.DataSchema;
+
+import java.util.*;
+
+public final class UriTemplate {
+
+  private UriTemplate(){
+    throw new UnsupportedOperationException();
+  }
+
+  public static List<String> extract(String path){
+    List<String> extracted = new ArrayList<>();
+    int n = path.length();
+    String s = "";
+    for (int i = 0; i < n; i++){
+      if (path.charAt(i) == '{'){
+        extracted.add(s);
+        s = "{";
+      }
+      else if (path.charAt(i)=='}'){
+        s = s +"}";
```

Comment:
Inconsistent use of whitespaces around operators, please review. This should be `s = s + "}";`